src/api/app/controllers/application_controller.rb

```diff
@@ -569,6 +569,17 @@ def set_response_format_to_xml
     request.format = :xml if request.format == :html
   end
 
+  def extract_user_public
```

Comment:
@mdeniz this method was already being using by `PublicController` hence why its in `ApplicationController`..

https://github.com/evanrolfe/open-build-service/blob/feature%2Fpublic_key_info_interface2/src/api/app/controllers/public_controller.rb#L21